### PR TITLE
[FEATURE] Support Ephemeral dashboards on the UI

### PR DESCRIPTION
**Pull request by AntoineThebaud**
# Description

Add the display of Ephemeral dashboards on the UI.
- the new tab on the project view is not displayed if there are no ephemeral dashboards registered.
- updating the TTL is possible from the project view, not the dashboard view. It's a design choice but there also a technical constraint (aka code genericity) preventing to allow modifying it from the JSON view in the current state.

NB: I made my best to share as much code as possible between Dashboards & Ephemeral Dashboards, I especially wanted to keep a single store & context for dashboards. It may still have room for improvement.

NB: Planned for later: add an option on the new duplicate action to make a dashboard copy ephemeral.

# Screenshots

image
image
2024-02-27_21h52_10

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/app/src/components/dialogs/DeleteEphemeralDashboardDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const { ephemeralDashboard, open, onClose, onSuccess } = props;
+  const { successSnackbar, exceptionSnackbar } = useSnackbar();
+  const deleteEphemeralDashboardMutation = useDeleteEphemeralDashboardMutation();
+
+  const handleSubmit = useCallback(() => {
+    return deleteEphemeralDashboardMutation.mutate(ephemeralDashboard, {
+      onSuccess: (deletedEphemeralDashboard: EphemeralDashboardResource) => {
+        successSnackbar(
+          `Ephemeral dashboard ${getDashboardExtendedDisplayName(
+            deletedEphemeralDashboard as unknown as DashboardResource
```
[the remark this answers is not in the document]
Comment (AntoineThebaud) [reply ui/app/src/components/dialogs/DeleteEphemeralDashboardDialog.tsx:45]: Yes I was hesitating to do so, 'will proceed then

File: ui/app/src/components/EphemeralDashboardList/EphemeralDashboardDataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+const MemoizedColumnHeaders = memo(GridColumnHeaders);
+
+export interface Row {
+  project: string;
+  name: string;
+  displayName: string;
+  version: number;
+  createdAt: string;
+  updatedAt: string;
+  expireAt: Date;
```
[the remark this answers is not in the document]
Comment (AntoineThebaud) [reply ui/app/src/components/EphemeralDashboardList/EphemeralDashboardDataGrid.tsx:44]: yes I will change for consistency

File: ui/core/src/model/ephemeraldashboard.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  kind: 'EphemeralDashboard';
+  metadata: ProjectMetadata;
+  spec: EphemeralDashboardSpec;
+}
+
+export interface EphemeralDashboardSpec extends DashboardSpec {
+  ttl: DurationString;
+}
+
+export interface EphemeralDashboardSelector extends DashboardSelector {
```
[the remark this answers is not in the document]
Comment (AntoineThebaud) [reply ui/core/src/model/ephemeraldashboard.ts:28]: ah yes forgot to change it, my bad

File: ui/app/src/components/GridToolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import { Stack } from '@mui/material';
+import {
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarFilterButton,
+  GridToolbarQuickFilter,
+} from '@mui/x-data-grid';
+
+export function GridToolbar() {
```
[the remark this answers is not in the document]
Comment (AntoineThebaud) [reply ui/app/src/components/GridToolbar.tsx:22]: I'll try to factorize further all this data grid part later, but for now this part was a quick win :D

**Review by Nexucis — APPROVED** [review]
noice !

**Pull request merged**